### PR TITLE
Adds Gigaton Hammer and Salt Cure

### LINKED ISSUE #2496: MOVE_SALT_CURE Effect

### LINKED ISSUE #2508: MOVE_GIGATON_HAMMER Effect

### LINKED ISSUE #3272: Pollen Puff is still a Ballistic Move in Gen 9
### Description

At the time, Bulbapedia had it marked as it losing its flag, but now it's been corrected, likely thanks to testing in Smogon's research thread.

Source: https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9498031

### Version

upcoming (Edge)

### Upcoming Version

1513443c393a4bd8575b5be1f3a3727bb9658ae5

### Discord contact info

AsparagusEduardo

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Closes #2508
Closes #2496
Fixes #3272

I added a Status_Freeze animation to the residual damage of Salt Cure to make it more obvious. 

Full credit goes to Lunos. I only did some clean up and added tests.

### CODE REVIEW

File: src/battle_ai_main.c
Comment (ghoulslash) [inline src/battle_ai_main.c:4960]: missing a break;

**Review by ghoulslash — CHANGES_REQUESTED** [review]
Otherwise looks good

File: src/battle_ai_main.c
Comment (AlexOn1ine) [reply src/battle_ai_main.c:4960]: added

File: src/battle_util.c
Comment (Skeli789) [inline src/battle_util.c:1904]: It functions the same as Torment, so there's really no need for an extra timer.

Comment (AlexOn1ine) [reply src/battle_util.c:1904]: Would this work? Torment applies to a battler so no move can be used twice in a row.

Comment (LOuroboros) [reply src/battle_util.c:1904]: > It functions the same as Torment, so there's really no need for an extra timer.

It kinda doesn't though. Torment affects a battler and forbids them from using X or Y move twice in a row until they switch out, and the way that's accomplished is by branding the battler in question with a full status flag, `STATUS2_TORMENT`.

Gigaton Hammer has a cooldown period essentially; you can't use that specific move twice in a row, that's why the timer works just fine instead of wasting a full status flag on it for no good reason :eyes: ...

Comment (Skeli789) [reply src/battle_util.c]: Alex understood what I meant. With a timer its not implemented correctly. See [this post](https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9412011) explaining why. With the timer implementation, a Pokémon would be allowed to select it again after being asleep which is not correct.

Comment (AlexOn1ine) [reply src/battle_util.c]: Removed the Gigaton Hammer timer.

Comment (Skeli789) [reply src/battle_util.c]: Looks good now, just do a test to see if it should be gLastMoves or gLastResultingMoves. gLastMoves may not let it be selected after being upgraded to a Z-Move.

Comment (Skeli789) [reply src/battle_util.c]: You also may want to implement the new Blood Moon attack while you have this open.

Comment (AlexOn1ine) [reply src/battle_util.c]: > gLastResultingMoves

Yep, good call, `gLastResultingMoves` should be used.

Comment (AsparagusEduardo) [reply src/battle_util.c]: > You also may want to implement the new Blood Moon attack while you have this open.

2-week spoiler policy 👀

**Comment by AsparagusEduardo** [comment]
Please merge with the latest upcoming to fix the CI issue.

**Comment by AsparagusEduardo** [comment]
Please merge with latest upcoming to fix the CI issue.

**Comment by AlexOn1ine** [comment]
> Please merge with latest upcoming to fix the CI issue.

done

File: test/battle/move_effect/salt_cure.c
Comment (AsparagusEduardo) [inline test/battle/move_effect/salt_cure.c:25]: Based on the identation, I think you wanted to add braces here.

File: src/battle_util.c
Comment (AlexOn1ine) [reply src/battle_util.c]: > > You also may want to implement the new ___________ attack while you have this open.
> 
> 2-week spoiler policy 👀

In this case I would say that ghoul should decide whether to wait those 2 weeks for the move in question or merge before it.

Comment (AsparagusEduardo) [reply src/battle_util.c]: > In this case I would say that ghoul should decide whether to wait those 2 weeks for the move in question or merge before it.

If we set Tackle with `EFFECT_GIGATON_HAMMER` and then using one after the other works, the effect itself should be ready, and in 2 weeks we only need to add the new move data.

Comment (AlexOn1ine) [reply src/battle_util.c]: Yeah, already checked. Works fine with 2 moves of the same effect on a moveset.

File: test/battle/move_effect/salt_cure.c
Comment (AlexOn1ine) [reply test/battle/move_effect/salt_cure.c:25]: Fixed.

**Review by AsparagusEduardo — APPROVED** [review]

**Comment by AsparagusEduardo** [comment]
Alright, just need to wait for @ghoulslash's review.

**Pull request merged**
